packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -491,46 +491,61 @@ export function authorizationServiceBuilder(
         showUsers: true,
       };
     },
-    async addClientUser(
+    async addClientUsers(
       {
         clientId,
-        userId,
+        userIds,
         authData,
       }: {
         clientId: ClientId;
-        userId: UserId;
+        userIds: UserId[];
         authData: AuthData;
       },
       correlationId: string,
       logger: Logger
     ): Promise<{ client: Client; showUsers: boolean }> {
-      logger.info(`Binding client ${clientId} with user ${userId}`);
+      logger.info(`Binding client ${clientId} with user ${userIds.join(",")}`);
       const client = await retrieveClient(clientId, readModelService);
       assertOrganizationIsClientConsumer(authData.organizationId, client.data);
-      await assertUserSelfcareSecurityPrivileges({
-        selfcareId: authData.selfcareId,
-        requesterUserId: authData.userId,
-        consumerId: authData.organizationId,
-        selfcareV2InstitutionClient,
-        userIdToCheck: userId,
-        correlationId,
+
+      await Promise.all(
+        userIds.map((userId) =>
+          assertUserSelfcareSecurityPrivileges({
+            selfcareId: authData.selfcareId,
+            requesterUserId: authData.userId,
+            consumerId: authData.organizationId,
+            selfcareV2InstitutionClient,
+            userIdToCheck: userId,
+            correlationId,
+          })
+        )
+      );
+
+      userIds.forEach((userId) => {
+        if (client.data.users.includes(userId)) {
+          throw clientUserAlreadyAssigned(clientId, userId);
+        }
       });
-      if (client.data.users.includes(userId)) {
-        throw clientUserAlreadyAssigned(clientId, userId);
-      }
+
+      const uniqueUserIds = Array.from(new Set(userIds));
       const updatedClient: Client = {
         ...client.data,
-        users: [...client.data.users, userId],
       };
 
-      await repository.createEvent(
-        toCreateEventClientUserAdded(
-          userId,
-          updatedClient,
-          client.metadata.version,
-          correlationId
-        )
-      );
+      for (const [index, userId] of uniqueUserIds.entries()) {
+        // eslint-disable-next-line functional/immutable-data
+        updatedClient.users.push(userId);
+
+        await repository.createEvent(
+          toCreateEventClientUserAdded(
+            userId,
+            updatedClient,
+            client.metadata.version + index,
+            correlationId
+          )
+        );
+      }
```

Comment:
I would have expected this to happen in the same transaction, am I wrong?

Reply:
You're right! didn't know there was already an event repository's method that could do that. Thanks